server/src/main/java/org/elasticsearch/index/mapper/IndexTimeScriptContext.java

```diff
@@ -0,0 +1,256 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.index.mapper;
+
+import org.apache.lucene.analysis.Analyzer;
+import org.apache.lucene.index.BinaryDocValues;
+import org.apache.lucene.index.DocValuesType;
+import org.apache.lucene.index.FieldInfo;
+import org.apache.lucene.index.FieldInfos;
+import org.apache.lucene.index.Fields;
+import org.apache.lucene.index.IndexOptions;
+import org.apache.lucene.index.IndexableField;
+import org.apache.lucene.index.LeafMetaData;
+import org.apache.lucene.index.LeafReader;
+import org.apache.lucene.index.LeafReaderContext;
+import org.apache.lucene.index.NumericDocValues;
+import org.apache.lucene.index.PointValues;
+import org.apache.lucene.index.SortedDocValues;
+import org.apache.lucene.index.SortedNumericDocValues;
+import org.apache.lucene.index.SortedSetDocValues;
+import org.apache.lucene.index.StoredFieldVisitor;
+import org.apache.lucene.index.Terms;
+import org.apache.lucene.index.memory.MemoryIndex;
+import org.apache.lucene.util.Bits;
+import org.elasticsearch.common.CheckedConsumer;
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.index.fielddata.IndexFieldDataCache;
+import org.elasticsearch.indices.breaker.NoneCircuitBreakerService;
+import org.elasticsearch.search.lookup.SearchLookup;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+public class IndexTimeScriptContext {
+
+    public final SearchLookup searchLookup;
+    public final LeafReaderContext leafReaderContext;
+    public final ParseContext pc;
+    private final Map<String, MapperScript> mapperScripts = new HashMap<>();
+
+    public static void executePostParsePhases(MappingLookup lookup, ParseContext parseContext) throws IOException {
+        if (lookup.getPostParsePhases().isEmpty()) {
+            return;
+        }
+        IndexTimeScriptContext context = new IndexTimeScriptContext(lookup, parseContext);
+        context.executePostParse();
+    }
+
+    private IndexTimeScriptContext(MappingLookup mappingLookup, ParseContext pc) {
+        this.searchLookup = new SearchLookup(
+            mappingLookup::getFieldType,
+            (ft, s) -> ft.fielddataBuilder("", s).build(
```

Comment:
This is a TODO - we should be able to get the index name from somewhere and pass it through here.